cpp/ql/src/jsf/4.17 Types/AV Rule 147.ql

```diff
@@ -24,8 +24,8 @@ class InvalidFloatCastExpr extends Expr {
 
   InvalidFloatCastExpr() {
     exists(Type src, Type dst |
-      src = this.getUnderlyingType().getUnspecifiedType() and
-      dst = this.getActualType().getUnderlyingType().getUnspecifiedType() and
+      src = this.getUnspecifiedType() and
+      dst = this.getActualType().getUnspecifiedType() and
```

Comment:
Next, you can replace `.getActualType().getUnspecifiedType()` with `.getFullyConverted().getUnspecifiedType()`.